internal/uri.go

```diff
@@ -0,0 +1,45 @@
+package internal
+
+import (
+	"net/url"
+	"os"
+	"strings"
+)
+
+// https://buildpacks.io/docs/reference/config/package-config/#dependencies-list-optional
+func IsArchive(dependency PackageConfigDependency) bool {
+	if IsCnbRegistry(dependency) {
```

Comment:
I think is better to do something like what you do on the isDirectory. To look if the file exists and is an archive. After you can check if the file extension is within a range of suffixes